app/lib/event_store/event.rb

```diff
@@ -66,6 +66,35 @@ def load(str)
 
     scope :stale, -> { where.has { created_at <= TTL.ago } }
 
+    class EventRollbackError < StandardError
+      include Bugsnag::MetaData
+
+      def initialize(error:, event:)
+        event_details = event.to_h.as_json
+        self.bugsnag_meta_data = { event: event_details }
+
+        super "Error raised trying to roll back #{event}: #{self}, event details: #{event_details}"
+        set_backtrace error.backtrace
+      end
+    end
+
+    # Since Rails 6.1 the EventStore::Event records (among others) are added to the
+    # list of the objects, belonging to a current transaction, and #rolledback! is called on them
+    # if the transaction is rolled back on the DB.
+    # The problem is that this triggers the Event's `load` and `deserialize` (to restore the original state), and if
+    # the object referenced in the event via GlobalID does not exist, the deserialization of the event will fail.
+    # This causes some tests (that have explicit transactions and rollbacks) to fail, and can potentially cause issues
+    # in production as well.
+    def rolledback!(*)
+      super
```

Comment:
my original suggestion was to skip this and only raise an error if `has_transactional_callbacks?` returns true. If it is not true, then we assue that it is safe to skip that call.

I know presently it is true. But I don't know what somebody can do in the future. So this is like a safety check for what we expect in the long term.

Reply:
I am not sure I understand your comment above @akostadinov ... 

Can you maybe suggest how you want this to be changed?

Reply:
Actually I didn't understand your code. I added a comment suggestion.

The only thing is probably we need to add a test about it. Maybe by mocking `has_transactional_callbacks?`

Reply:
@akostadinov what about this: https://github.com/3scale/porta/pull/3705/commits/b7da17c98df6fee5aac17c3eab33e296786c8030